lib/utils/joinValues.ts

```diff
@@ -0,0 +1,48 @@
+import { stringifyValue } from './stringify.ts';
+import type { Primitive } from 'zod';
```

Comment:
question: do we have an eslint rule that moves 3rd party imports on top, and keeps the inner/relative ones below?